### PR TITLE
Glob expansion

### PR DESCRIPTION
**Pull request by catilac**
I wrote a first attempt here. I think I could leverage the stuff in `util.go` for the environment variables, but decided not to. Is this direction reasonable? I'm trying to figure out how robust it all needs to be.

### CODE REVIEW

**Comment by progrium** [comment]
It's a start! One thing I noticed is that this only supports variables that are standalone arguments. We could better support bash style variables (including modifiers) using @mgood's https://github.com/mgood/go-posix

**Comment by catilac** [comment]
ah cool! thanks for looking. i'll take a look at that library!

**Comment by catilac** [comment]
references issue #41

**Comment by progrium** [comment]
Just getting to this. At a glance it looks good. @Parzival-3141 wanna review and merge?

**Review by Parzival-3141 — APPROVED** [review]
This is a good base to work off of. Thank you!

File: shell/preprocessor.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			}
+			// add expanded glob into result
+			for _, match := range matches {
+				result = append(result, match)
+			}
+			continue
+		}
+
+		if arg[0] == '$' {
+			val, err := posix.Expand(arg, mapping)
```
Comment (mgood) [inline shell/preprocessor.go:37]: Oh I meant to look at this sooner, but I believe you can use `ExpandEnv` here instead to handle the env var substitution

Comment (catilac) [reply shell/preprocessor.go:37]: Oh! I'll make another PR to address that. Thanks @mgood

Comment (Parzival-3141) [reply shell/preprocessor.go:37]: I think in this case we want control over where the environment vars are coming from, so passing in `env` would be preferable.

**Pull request merged**